pkg/snmp/traps/formatter.go

```diff
@@ -161,6 +161,28 @@ func NormalizeOID(value string) string {
 	return strings.TrimLeft(value, ".")
 }
 
+// parseValue checks to see if the variable has a mapping in an enum and
+// returns the mapping if it exists, otherwise returns the value unchanged
+func parseValue(variable trapVariable, varMetadata VariableMetadata) interface{} {
+	if variable.VarType == "integer" {
+		if i, ok := variable.Value.(int); !ok {
+			// TODO(ken): should we do more here? Should we even handle this scenario?
+			log.Debugf("unable to parse value of type \"integer\": %+v", variable.Value)
+		} else if len(varMetadata.Enumeration) > 0 {
+			// if we find a mapping set it and return
+			value := varMetadata.Enumeration[i]
+			if value == "" {
```

Comment:
Can this really happen? Maybe we should prevent this when loading the database?

Reply:
Aah sorry, I was under the impression that `varMetadata.Enumeration[i]` would panic if it was called with a non-existing index, but you're right that in go it will return the string null value so `""`.
Wondering if we should do an explicit check for presence instead with `value, ok := varMetadata.Enumeration[i]` for clarity instead. What do you think?

Reply:
Ahh gotcha. This is a great question and I've gone back and forth on it so many times for maps. I used to be hard core in favor of `value, ok :=` and then I started working on a team that hated doing that so I switched.

I'm good with being explicit here.